engine/src/main/java/org/terasology/engine/module/StandardModuleExtension.java

```diff
@@ -47,17 +59,34 @@ public String getKey() {
     }
 
     public static boolean isServerSideOnly(Module module) {
-        Boolean serverSideOnly = module.getMetadata().getExtension(SERVER_SIDE_ONLY.getKey(), Boolean.class);
-        return serverSideOnly != null && serverSideOnly;
+        return getBooleanExtension(module, SERVER_SIDE_ONLY);
     }
 
     public static boolean isGameplayModule(Module module) {
-        Boolean isGameplay = module.getMetadata().getExtension(IS_GAMEPLAY.getKey(), Boolean.class);
-        return isGameplay != null && isGameplay;
+        return getBooleanExtension(module, IS_GAMEPLAY);
     }
 
     public static SimpleUri getDefaultWorldGenerator(Module module) {
         String ext = module.getMetadata().getExtension(DEFAULT_WORLD_GENERATOR.getKey(), String.class);
         return ext != null ? new SimpleUri(ext) : null;
     }
+
+    private static boolean getBooleanExtension(Module module, StandardModuleExtension ext) {
+        Boolean result = module.getMetadata().getExtension(ext.getKey(), Boolean.class);
+        return result != null && result;
+    }
+
+	public static Set<StandardModuleExtension> booleanPropertySet() {
+		Set<StandardModuleExtension> booleanPropertySet = EnumSet.noneOf(StandardModuleExtension.class);
+		for (StandardModuleExtension standardModuleExtension : values()) {
+			if (standardModuleExtension.getValueType().equals(Boolean.class)) {
+				booleanPropertySet.add(standardModuleExtension);
+			}
+		}
+		return booleanPropertySet;
+	}
+
+	public boolean isProvidedBy(Module module) {
```

Comment:
I had a terrible time coming up with a name for this method.   Maybe someone else can think of a better one.

Reply:
Seems a fair enough name to me. Flows well in its one usage